datafusion/datasource/src/file_groups.rs

```diff
@@ -354,6 +361,115 @@ impl FileGroupPartitioner {
     }
 }
 
+/// Represents a group of partitioned files that'll be processed by a single thread.
+/// Maintains optional statistics across all files in the group.
+#[derive(Debug, Clone)]
+pub struct FileGroup {
+    /// The files in this group
+    pub files: Vec<PartitionedFile>,
+    /// Optional statistics for all files in the group
```

Comment:
I think we could make it clearer here that the statistics are for the combined data in the files. Something like

```suggestion
    /// Optional statistics for the data across all files in the group
```